src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/OpenSslX509CertificateReader.cs

```diff
@@ -632,6 +652,26 @@ public ICertificatePal CopyWithPrivateKey(ECDsa privateKey)
             }
         }
 
+        public ICertificatePal CopyWithPrivateKey(ECDiffieHellman privateKey)
+        {
+            ECDiffieHellmanOpenSsl? typedKey = privateKey as ECDiffieHellmanOpenSsl;
+
+            if (typedKey != null)
+            {
+                return CopyWithPrivateKey((SafeEvpPKeyHandle)typedKey.DuplicateKeyHandle());
+            }
+
+            ECParameters ecParameters = privateKey.ExportParameters(true);
+
+            using (PinAndClear.Track(ecParameters.D!))
+            using (typedKey = new ECDiffieHellmanOpenSsl())
+            {
+                typedKey.ImportParameters(ecParameters);
+
+                return CopyWithPrivateKey((SafeEvpPKeyHandle)typedKey.DuplicateKeyHandle());
```

Comment:
Why is this cast needed? That looks like the return type of DuplicateKeyHandle to me...

Reply:
It's not; this code was "inspired" from the ECDSA `CopyWithPrivateKey` which also has the redundant cast. I'll remove all of them.